daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionVersion.scala

```diff
@@ -6,51 +6,53 @@ package transaction
 
 import com.daml.lf.data.ImmArray
 import com.daml.lf.language.LanguageVersion
+import com.daml.lf.transaction.TransactionVersion.assignNodeVersion
 import com.daml.lf.value.{Value, ValueVersion}
-import scalaz.NonEmptyList
 
 import scala.collection.immutable.HashMap
 
-final case class TransactionVersion(protoValue: String)
+sealed abstract class TransactionVersion private (val protoValue: String, private val index: Int)
+    extends Product
+    with Serializable
 
 /**
   * Currently supported versions of the DAML-LF transaction specification.
   */
-object TransactionVersion
-    extends LfVersions(versionsAscending =
-      NonEmptyList(new TransactionVersion("10"), new TransactionVersion("dev")))(
-      _.protoValue,
-    ) {
+object TransactionVersion {
 
-  private[lf] implicit val Ordering: Ordering[TransactionVersion] = mkOrdering
+  case object V10 extends TransactionVersion("10", 0)
```

Comment:
Do we really want `0` here as the index or rather `10`?

Reply:
good point.